CMakeLists.txt

```diff
@@ -156,6 +156,12 @@ include_directories(SYSTEM ${LZ4_INCLUDE_DIR})
 add_library(lz4static STATIC IMPORTED)
 set_target_properties(lz4static PROPERTIES IMPORTED_LOCATION ${LZ4_STATIC_LIB})
 
+## ZLIB
+find_package(ZLIB REQUIRED)
+include_directories(SYSTEM ${ZLIB_INCLUDE_DIRS})
+add_library(z STATIC IMPORTED)
+set_target_properties(z PROPERTIES IMPORTED_LOCATION ${ZLIB_LIBRARIES})
```

Comment:
cmake does ship with `FindZLIB.cmake`, but it uses `ZLIB_ROOT` for custom installation directories. I would prefer to ship this and change it to use `ZLIB_HOME` so we aren't dependent on the cmake file that ships with cmake

https://github.com/Kitware/CMake/blob/master/Modules/FindZLIB.cmake